pkg/kotsadmsnapshot/types/types.go

```diff
@@ -59,29 +60,38 @@ type App struct {
 	AppIconURI string `json:"iconUri"`
 }
 
-// ReplicatedBackup holds both the infrastructure and app backups for an EC cluster
-type ReplicatedBackup struct {
-	Name string `json:"name"`
-	// number of backups expected to exist for the ReplicatedBackup to be considered complete
-	ExpectedBackupCount int      `json:"expectedBackupCount"`
-	Backups             []Backup `json:"backups"`
-}
+// BackupStatus represents the status of a backup
+type BackupStatus string
 
+const (
+	// BackupStatusInProgress indicates that the backup is currently in progress
+	BackupStatusInProgress BackupStatus = "InProgress"
+	// BackupStatusCompleted indicates that the backup has been completed successfully
+	BackupStatusCompleted BackupStatus = "Completed"
+	// BackupStatusFailed indicates that the backup has failed
+	BackupStatusFailed BackupStatus = "Failed"
+	// BackupStatusDeleting indicates that the backup is being deleted
+	BackupStatusDeleting BackupStatus = "Deleting"
+)
+
+// Backup represnts a replicated backup working as an abstraction layer between Replicated and
+// Velero backups. These can be either infrastructure/instance, app backups or both.
 type Backup struct {
-	Name               string     `json:"name"`
-	Status             string     `json:"status"`
-	Trigger            string     `json:"trigger"`
-	AppID              string     `json:"appID"`    // TODO: remove with app backups
-	Sequence           int64      `json:"sequence"` // TODO: remove with app backups
-	StartedAt          *time.Time `json:"startedAt,omitempty"`
-	FinishedAt         *time.Time `json:"finishedAt,omitempty"`
-	ExpiresAt          *time.Time `json:"expiresAt,omitempty"`
-	VolumeCount        int        `json:"volumeCount"`
-	VolumeSuccessCount int        `json:"volumeSuccessCount"`
-	VolumeBytes        int64      `json:"volumeBytes"`
-	VolumeSizeHuman    string     `json:"volumeSizeHuman"`
-	SupportBundleID    string     `json:"supportBundleId,omitempty"`
-	IncludedApps       []App      `json:"includedApps,omitempty"`
+	Name            string       `json:"name"`
+	Status          BackupStatus `json:"status"`
+	Trigger         string       `json:"trigger"`
+	AppID           string       `json:"appID"`    // TODO: remove with app backups
+	Sequence        int64        `json:"sequence"` // TODO: remove with app backups
+	StartedAt       *time.Time   `json:"startedAt,omitempty"`
+	FinishedAt      *time.Time   `json:"finishedAt,omitempty"`
+	ExpiresAt       *time.Time   `json:"expiresAt,omitempty"`
+	SupportBundleID string       `json:"supportBundleId,omitempty"`
+	IncludedApps    []App        `json:"includedApps,omitempty"`
+	// number of velero backups expected to exist for the Backup to be considered done
+	ExpectedBackupCount int `json:"expectedBackupCount"`
+	// number of velero backups that actually exist
+	BackupCount int `json:"backupCount"`
```

Comment:
With us no longer having the `ReplicatedBackup` struct I believe these 2 fields end up being useful, specifically when we convey `Failed` states it's possible for the client to check if the counts match. They weren't part of the design doc proposal but I ended up adding them as I worked through the code.